### ISSUE TITLE
smc++ installation issue

### ISSUE THREAD

**Issue by Gogol83** [popgenmethods/smcpp#84]
Hi,

I have followed the instruction given at [https://github.com/popgenmethods/smcpp#installation-instructions](url) and installed smc++ in a Python root environment on a HPC server. However, when I am running smc++, the following message is being returned:

```Traceback (most recent call last):
  File "/home/users/ntu/nidhan/.conda/envs/my_root/bin/smc++", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/home/users/ntu/nidhan/.conda/envs/my_root/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3095, in <module>
    @_call_aside
  File "/home/users/ntu/nidhan/.conda/envs/my_root/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3079, in _call_aside
    f(*args, **kwargs)
  File "/home/users/ntu/nidhan/.conda/envs/my_root/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3108, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/home/users/ntu/nidhan/.conda/envs/my_root/lib/python3.6/site-packages/pkg_resources/__init__.py", line 570, in _build_master
    ws.require(__requires__)
  File "/home/users/ntu/nidhan/.conda/envs/my_root/lib/python3.6/site-packages/pkg_resources/__init__.py", line 888, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/users/ntu/nidhan/.conda/envs/my_root/lib/python3.6/site-packages/pkg_resources/__init__.py", line 774, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'attrs>=17.2.0' distribution was not found and is required by smcpp
```

I have tried uninstalling `attrs` and reinstalling it from `terhorst` channle, but the issue remains.

Below is the return for `conda list`:

```
# packages in environment at /home/users/ntu/nidhan/.conda/envs/my_root:
#
ca-certificates           2018.03.07                    0    anaconda
cloog                     0.18.0                        0    anaconda
gcc                       4.8.5                         7    anaconda
gmp                       6.1.2                h6c8ec71_1    anaconda
isl                       0.12.2                        0    anaconda
krb5                      1.13.2                        0    anaconda
libgcc-ng                 7.2.0                hdf63c60_3    anaconda
libstdcxx-ng              7.2.0                hdf63c60_3    anaconda
mkl                       2017.0.3                      0    anaconda
mpc                       1.0.3                hec55b23_5    anaconda
mpfr                      3.1.5                h11a74b3_2    anaconda
openssl                   1.0.2l               h077ae2c_5    anaconda
bcftools                  1.7                           0    bioconda
gnuplot                   5.0.4              ncurses5.9_1    bioconda
htslib                    1.7                           0    bioconda
pysam                     0.14.1          py36_htslib1.7_0    bioconda
samtools                  1.7                           2    bioconda
appdirs                   1.4.3                      py_0    conda-forge
blas                      1.1                    openblas    conda-forge
bzip2                     1.0.6                         1    conda-forge
certifi                   2018.4.16                py36_0    conda-forge
curl                      7.52.1                        0    conda-forge
cycler                    0.10.0                   py36_0    conda-forge
cython                    0.28.2                   py36_0    conda-forge
dbus                      1.11.0                        0    conda-forge
expat                     2.2.5                         0    conda-forge
fontconfig                2.12.6                        0    conda-forge
freetype                  2.8.1                         0    conda-forge
gettext                   0.19.8.1                      0    conda-forge
giflib                    5.1.4                         0    conda-forge
glib                      2.55.0                        0    conda-forge
gsl                       2.4             blas_openblas_0  [blas_openblas]  conda-forge
gst-plugins-base          1.8.0                         0    conda-forge
gstreamer                 1.8.0                         1    conda-forge
icu                       58.2                          0    conda-forge
jpeg                      9b                            2    conda-forge
kiwisolver                1.0.1                    py36_1    conda-forge
libffi                    3.2.1                         3    conda-forge
libgd                     2.2.5                         3    conda-forge
libiconv                  1.15                          0    conda-forge
libpng                    1.6.34                        0    conda-forge
libtiff                   4.0.9                         0    conda-forge
libwebp                   0.5.2                         7    conda-forge
libxcb                    1.13                          0    conda-forge
libxml2                   2.9.4                         4    conda-forge
matplotlib                2.2.2                    py36_1    conda-forge
ncurses                   5.9                          10    conda-forge
numpy                     1.13.1          py36_blas_openblas_201  [blas_openblas]  conda-forge
openblas                  0.2.20                        7    conda-forge
pandas                    0.23.0                   py36_1    conda-forge
pcre                      8.41                          1    conda-forge
pip                       9.0.3                    py36_0    conda-forge
pyparsing                 2.2.0                    py36_0    conda-forge
pyqt                      5.6.0                    py36_5    conda-forge
python                    3.6.5                         1    conda-forge
python-dateutil           2.7.3                      py_0    conda-forge
pytz                      2018.4                     py_0    conda-forge
qt                        5.6.2                         7    conda-forge
readline                  7.0                           0    conda-forge
scipy                     1.1.0           py36_blas_openblas_200  [blas_openblas]  conda-forge
setuptools                39.2.0                   py36_0    conda-forge
setuptools_scm            2.1.0                    py36_0    conda-forge
sip                       4.18                     py36_1    conda-forge
six                       1.11.0                   py36_1    conda-forge
sqlite                    3.20.1                        2    conda-forge
tk                        8.6.7                         0    conda-forge
tornado                   5.0.2                    py36_0    conda-forge
tqdm                      4.23.3                     py_0    conda-forge
wheel                     0.31.0                   py36_0    conda-forge
wrapt                     1.10.11                  py36_0    conda-forge
xorg-libxau               1.0.8                         3    conda-forge
xorg-libxdmcp             1.1.2                         3    conda-forge
xz                        5.2.3                         0    conda-forge
zlib                      1.2.11                        0    conda-forge
libgcc                    5.2.0                         0  
libgfortran               3.0.0                         1  
ad                        1.3.2                    py36_0    terhorst
attrs                     17.2.0             pyh240c617_0    terhorst
smcpp                     1.14.0dev        py36h6bb024c_0    terhorst
```

And here, for `conda info`

```Current conda install:

             platform : linux-64
        conda version : 4.1.6
    conda-env version : 2.5.1
  conda-build version : 1.21.3
       python version : 3.5.2.final.0
     requests version : 2.10.0
     root environment : /app/anaconda/3  (read only)
  default environment : /home/users/ntu/nidhan/.conda/envs/my_root
     envs directories : /home/users/ntu/nidhan/.conda/envs
                        /home/users/ntu/nidhan/envs
                        /app/anaconda/3/envs
        package cache : /home/users/ntu/nidhan/.conda/envs/.pkgs
                        /home/users/ntu/nidhan/envs/.pkgs
                        /app/anaconda/3/pkgs
         channel URLs : https://conda.anaconda.org/ilan/linux-64/
                        https://conda.anaconda.org/ilan/noarch/
                        https://repo.continuum.io/pkgs/free/linux-64/
                        https://repo.continuum.io/pkgs/free/noarch/
                        https://repo.continuum.io/pkgs/pro/linux-64/
                        https://repo.continuum.io/pkgs/pro/noarch/
          config file : /home/users/ntu/nidhan/.condarc
         offline mode : False
    is foreign system : False
```

Could you please help me resolve this issue?

**Comment by terhorst** [comment]
Sorry, I don't know what the problem is.

**Comment by szpiech** [comment]
Did you ever resolve the issue @Gogol83 ? I'm having a similar problem after installing via conda on two different HPCs. I get a slight variation.

```smc++ plot -h
Traceback (most recent call last):
  File "~/anaconda3/bin/smc++", line 5, in <module>
    from pkg_resources import load_entry_point
  File "~/anaconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3250, in <module>
    @_call_aside
  File "~/anaconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3234, in _call_aside
    f(*args, **kwargs)
  File "~/anaconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3263, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "~/anaconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "~/anaconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "~/anaconda3/lib/python3.6/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'smcpp==1.3.9.dev2+ng00dfb92.d20170111' distribution was not found and is required by the application```

**Comment by michaelkarlcoleman** [comment]
For what it's worth, here is the install command that seemed to work for me:
```
conda create -n smcpp --override-channels -c defaults -c conda-forge -c terhorst -c bioconda smcpp 'attrs<19.2'
```

Note in particular that the `attrs` module API introduced a backward-incompatible change (hooray), and so the current `smcpp` conda install instructions probably no longer work as is.

**Comment by brianzhang01** [comment]
@szpiech I got the same issue on Fedora. Will report if I find a fix.
```
Traceback (most recent call last):
  File "/homes/bzhang/anaconda3/bin/smc++", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/homes/bzhang/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3241, in <module>
    @_call_aside
  File "/homes/bzhang/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3225, in _call_aside
    f(*args, **kwargs)
  File "/homes/bzhang/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 3254, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/homes/bzhang/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 583, in _build_master
    ws.require(__requires__)
  File "/homes/bzhang/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 900, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/homes/bzhang/anaconda3/lib/python3.7/site-packages/pkg_resources/__init__.py", line 786, in resolve
    raise DistributionNotFound(req, requirers)
pkg_resources.DistributionNotFound: The 'smcpp==1.3.9.dev2+ng00dfb92.d20170111' distribution was not found and is required by the application
```

**Comment by brianzhang01** [comment]
@szpiech I noticed that if I open the file `/homes/bzhang/anaconda3/bin/smcpp` there is a path for `smcpp==1.3.9.dev2+ng00dfb92.d20170111` (along with some easy_install logic). I presume one could try rewriting this path to 1.3.8, which seems like the actual version that was installed on my system.

However, what I ended up doing was doing a fresh conda environment with @michaelkarlcoleman 's command.
